plugins/kobolight.koplugin/main.lua

```diff
@@ -140,21 +140,22 @@ function KoboLight:onSwipe(_, ges)
         new_intensity = powerd.fl_intensity + delta_int
     elseif ges.direction == "south" then
         new_intensity = powerd.fl_intensity - delta_int
+    else
+        return false  -- don't consume swipe event if it's not matched
     end
-    if new_intensity ~= nil then
-        -- when new_intensity <=0, toggle light off
-        if new_intensity <=0 then
-            if powerd.is_fl_on then
-                powerd:toggleFrontlight()
-            end
-            self:onShowOnOff()
-        else -- general case
-            powerd:setIntensity(new_intensity)
-            self:onShowIntensity()
-        end
-        if self.view.footer_visible and self.view.footer.settings.frontlight then
-            self.view.footer:updateFooter()
+
+    -- when new_intensity <=0, toggle light off
+    if new_intensity <=0 then
+        if powerd.is_fl_on then
+            powerd:toggleFrontlight()
         end
+        self:onShowOnOff()
+    else -- general case
+        powerd:setIntensity(new_intensity)
+        self:onShowIntensity()
+    end
+    if self.view.footer_visible and self.view.footer.settings.frontlight then
+        self.view.footer:updateFooter()
     end
     return true
```

Comment:
I really think this is a bad design. Using event should be a better choice.
Though this does not relate to your change.